sklearn/linear_model/bayes.py

```diff
@@ -239,29 +253,17 @@ def fit(self, X, y, sample_weight=None):
             alpha_ = ((n_samples - gamma_ + 2 * alpha_1) /
                       (rmse_ + 2 * alpha_2))
 
-            # Check for convergence
-            if iter_ != 0 and np.sum(np.abs(coef_old_ - coef_)) < self.tol:
-                if verbose:
-                    print("Convergence after ", str(iter_), " iterations")
-                break
-            coef_old_ = np.copy(coef_)
-
+        if not converged:
+            warnings.warn("Optimization step did not converge. Increase the "
+                          "number of iterations.", ConvergenceWarning)
         self.n_iter_ = iter_ + 1
 
         # return regularization parameters and corresponding posterior mean,
-        # log marginal likelihood and posterior covariance
+        # objective function and posterior covariance
         self.alpha_ = alpha_
         self.lambda_ = lambda_
-        self.coef_, rmse_ = self._update_coef_(X, y, n_samples, n_features,
```

Comment:
I agree that you can remove this if converge = True, However if converge is False then you need to recompute the coef_ and scores_ that are based on the last value of alpha_ and lambda_

Reply:
It has been fixed.